crates/sui-storage/src/event_store/sql.rs

```diff
@@ -62,17 +67,31 @@ const INDEXED_COLUMNS: &[&str] = &[
 ];
 
 impl SqlEventStore {
-    /// Creates a new SQLite database for event storage
-    /// db_path may be a regular path starting with "/" or ":memory:" for in-memory database.
-    pub async fn new_sqlite(db_path: &str) -> Result<Self, EventStoreError> {
-        let pool = SqlitePool::connect(format!("sqlite:{}", db_path).as_str()).await?;
-        info!(db_path, "Created new SQLite EventStore");
+    /// Creates a new SQLite in-memory database, mostly for testing
+    pub async fn new_memory_only_not_prod() -> Result<Self, SuiError> {
```

Comment:
nit: maybe put this into an test-only `Ext`

Reply:
There might be some use cases for it that are not test-only, I think we should leave it

Reply:
No it doesn't matter which way you do it.